ldms/src/contrib/sampler/variorum_sampler/variorum_sampler.c

```diff
@@ -140,107 +127,178 @@ static int config(struct ldmsd_plugin *self, struct attr_value_list *kwl, struct
         return rc;
 }
 
 static int sample(struct ldmsd_sampler *self)
 {
         json_t *power_obj = NULL;
+        json_t *node_obj = NULL;
         int ret, socket;
 
-        if (!set) {
+        if (!set)
+        {
                 ovis_log(mylog, OVIS_LERROR, "plugin not initialized\n");
                 return EINVAL;
         }
 
         base_sample_begin(base);
 
-        // get variorum data
-        ret = variorum_get_node_power_json(&result_string);
-        if (ret != 0) {
+        // Get Variorum data
+        ret = variorum_get_power_json(&result_string);
+        if (ret != 0)
+        {
                 ovis_log(mylog, OVIS_LERROR, "unable to obtain JSON object data\n");
                 return EINVAL;
         }
 
         power_obj = json_loads(result_string, JSON_DECODE_ANY, NULL);
+        void *iter = json_object_iter(power_obj);
+        while (iter)
+        {
+                node_obj = json_object_iter_value(iter);
+                if (node_obj == NULL)
+                {
+                        printf("JSON object not found");
+                        exit(0);
+                }
+                /* The following should return NULL after the first call per our object. */
+                iter = json_object_iter_next(power_obj, iter);
+        }
 
-        double power_node = json_real_value(json_object_get(power_obj, "power_node_watts"));
-        double power_cpu, power_gpu, power_mem;
+        double power_node = -1.0;
+        double power_cpu = -1.0;
+        double power_gpu = -1.0;
+        double power_mem = -1.0;
+        int num_gpus_per_socket = -1;
+        char socketID[20];
+
+        // If we're on a GPU-only build, we don't have power_node_watts.
+        if (json_object_get(node_obj, "power_node_watts") != NULL)
+        {
+                power_node = json_real_value(json_object_get(node_obj, "power_node_watts"));
+        }
+
+        // If we're on a CPU-only build, we don't have num_gpus_per_socket
+        if (json_object_get(node_obj, "num_gpus_per_socket") != NULL)
+        {
+                num_gpus_per_socket = json_integer_value(json_object_get(node_obj,
+                                                                         "num_gpus_per_socket"));
+        }
 
-        // update each record
-        for(socket = 0; socket < nsockets; socket++) {
+        // Update each record
+        for (socket = 0; socket < nsockets; socket++)
+        {
+                // Node power is same on both sockets.
                 ldms_record_set_double(rec_idxs[socket], i_node, power_node);
-                power_cpu = json_real_value(json_object_get(power_obj, metric_names[socket]));
+
+                // Obtain Socket Object
+                snprintf(socketID, 20, "socket_%d", socket);
+                json_t *socket_obj = json_object_get(node_obj, socketID);
+                if (socket_obj == NULL)
+                {
+                        printf("Socket object not found!\n");
```

Comment:
Please don't use printf. Use the the ovis_log functions.

Reply:
Shall update this shortly, @morrone! 

We're debugging a segfault that has come up after my rebasing/conflict resolution (my last rebase was Feb 2, which was around the time we tested on Mutt successfully. 

The plugin loading system changed after that in March, from the commits [here](https://github.com/ovis-hpc/ldms/commit/e4aa21a42d6fe3a906e75408509a6b61f520e0c5#diff-685cc5c85b88cd3508c1de831060c7d95223513c9ebf25ce8d9e09c1b74c0532)). 
The Variorum plugin wasn't tested when this update went in -- so something may be broken here, or I may have to change something in my scripts that we tested in Feb. Trying to figure out what is causing the issue.

Reply:
@tpatki You might consider basing your PR on b4.4 first, and leave porting forward to main for a later task. We'll need to backport the plugin to b4.4 anyway.

Reply:
FYI, partially I suggest focusing on branch b4.4 because there are even bigger plugin API changes for main that will land any day now, and probably keep changing over the next few weeks.

Reply:
Noted @morrone. I will focus on b4.4 if I get to it this week. 
I'll be on vacation Apr 19 - May 1, so may have to revisit this and pick the right branch once I get back, depending on time.